src/basic/util.c

```diff
@@ -23,9 +27,36 @@ int saved_argc = 0;
 char **saved_argv = NULL;
 static int saved_in_initrd = -1;
 
+#if HAVE_XENCTRL
+static int xen_kexec_loaded(void) {
+        int ret = -1;
+
+        xc_interface *xch;
+
+        if (access("/proc/xen", F_OK) < 0)
+                return -1;
+
+        xch = xc_interface_open(NULL, NULL, 0);
+        if (!xch)
+                return -1;
```

Comment:
`return log_debug_errno(errno, "Failed to open xen control interface: %m)` (assuming errno is set)